### PR TITLE
1072 - [Instage] JS library fix

### PR DESCRIPTION
**Pull request by buckyroberts**
Closes https://github.com/MagnivOrg/prompt-layer-api/issues/1072

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -394,20 +396,45 @@ const cleaned_result = (results: any[]) => {
...
+      }
+
+      if ("content" in delta) {
+        response.content = `${response["content"]}${delta.content}`;
+      }
+    }
+
+    if (function_arguments || function_name) {
+      response.content = null;
+      response['function_call'] = {
```
[the remark this answers is not in the document]
Comment (Jped) [reply src/utils.ts:426]: if can be either `tool_calls` or `function_call` 
https://platform.openai.com/docs/api-reference/chat/object

I think whatever the user passes in we should use when we send it to us.

[the remark this answers is not in the document]
Comment (Jped) [reply src/utils.ts:426]: BTW this is the description of what delta can include: https://platform.openai.com/docs/api-reference/chat/streaming

[the remark this answers is not in the document]
Comment (Jped) [reply src/utils.ts:426]: if you notice function_call and tools_call the relevant information is nested differently

File: src/utils.ts [hunk 2]
```diff
@@ -378,14 +378,16 @@ const publishPromptTemplate = async (body: PublishPromptTemplate) => {
 };
 
 const cleaned_result = (results: any[]) => {
```
Comment (abubakarsohail) [inline src/utils.ts:380]: @buckyroberts can you also provide an example of how and where these changes will be applicable? 
Also, I think we should just catch all the streaming responses and send them to the server for post processing. JavaScript sdk mimics python sdk and I believe the same issue would be present there as well.

**Review by Jped — APPROVED** [review]

**Pull request merged**